services/jitpack/jitpack-downloads.service.js

```diff
@@ -1,14 +1,83 @@
 'use strict'
 
-const { deprecatedService } = require('..')
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+const { downloadCount } = require('../color-formatters')
+const { metric } = require('../text-formatters')
 
-module.exports = deprecatedService({
-  route: {
-    base: 'jitpack',
-    pattern: ':period(dw|dm)/:various*',
+const schema = Joi.object({
+  week: nonNegativeInteger,
+  month: nonNegativeInteger,
+}).required()
+
+const variantMap = {
+  dw: {
+    api_field: 'week',
+    suffix: '/week',
+  },
+  dm: {
+    api_field: 'month',
+    suffix: '/month',
   },
-  message: 'temporarily unavailable',
-  label: 'downloads',
-  category: 'downloads',
-  dateAdded: new Date('2019-05-27'),
-})
+}
+
+module.exports = class JitpackDownloads extends BaseJsonService {
+  static get category() {
+    return 'downloads'
+  }
+
+  static get route() {
+    return {
+      base: 'jitpack',
+      pattern:
+        ':variant(dw|dm)/:vcs(github|bitbucket|gitlab|gitee)/:user/:repo',
```

Comment:
@chris48s & @paulmelnikow, did we land on using `variant` for time period/intervals in #3495? The original (pre-#3493 deprecation) version of this badge used `period`  but changed this to `variant` based on #3495

Reply:
I think where it is specifically a timeframe like `dm|dw`, we've got a mix of `interval` and `period` in various places, but I think Paul is right that `interval` is more descriptive and we should standardise on that.